src/tests/unit/audit_remediation/test_audit_remediation_base.py

```diff
@@ -309,6 +312,27 @@ def test_result_query_export_file(cbcsdk_mock):
         os.remove(tempfile)
 
 
+@pytest.mark.parametrize("ref_url, func_raises, get_job", [
+    ('https://example.com/jobs/v1/orgs/test/jobs/12345', does_not_raise(), True),
+    ('https://example.com/jobs/v1/orgs/test/jobs/NOTVALID', pytest.raises(ApiError), False),
+    (None, pytest.raises(ApiError), False),
+])
+def test_result_query_async_export(cbcsdk_mock, ref_url, func_raises, get_job):
+    """Tests getting a Job from the SDK to asynchronously export a query's results."""
+    cbcsdk_mock.mock_request("POST", "/livequery/v1/orgs/test/runs/run_id/results/_search?format=csv&async=true",
+                             {"ref_url": ref_url})
+    if get_job:
+        cbcsdk_mock.mock_request('GET', '/jobs/v1/orgs/test/jobs/12345', JOB_DETAILS_1)
+    api = cbcsdk_mock.api
+    result_query = api.select(Result).run_id("run_id")
+    with func_raises:
+        job = result_query.async_export()
```

Comment:
What is the intended code that a developer should write for waiting on the job to complete? Should this be managed through an async function where they do results() and it will block or is the expectation that they use a while loop until the status is completed maybe adding a sleep timer to prevent API overload

Reply:
In my little piece of test code, I did something like this:
```
progress_data = job.get_progress()
while progress_data[1] < progress_data[0]:
    print("Waiting for job to complete...")
    progress_data = job.get_progress()
```
You could throw a sleep() in the middle of that if you wanted.  Actually, I found that I had to put a small sleep() in _before_ doing this, as the progress call would fail if I tried calling it _immediately_ after creating the job.

Reply:
This is a decent amount of complexity which we could build into a python future option. Can we add that as an option here?

Reply:
Yes, via the new `Job` method I've just added, `await_completion()`. It returns a `Future` similar to an async query.